core/src/main/scala/org/apache/spark/storage/BlockManagerMasterHeartbeatEndpoint.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.storage
+
+import scala.collection.mutable
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.rpc.{IsolatedRpcEndpoint, RpcCallContext, RpcEnv}
+import org.apache.spark.storage.BlockManagerMessages.{BlockManagerHeartbeat, StopBlockManagerMaster}
+
+/**
+ * Separate heartbeat out of BlockManagerMasterEndpoint due to performance consideration.
+ */
+private[spark] class BlockManagerMasterHeartbeatEndpoint(
+    override val rpcEnv: RpcEnv,
+    isLocal: Boolean,
+    blockManagerInfo: mutable.Map[BlockManagerId, BlockManagerInfo])
+  extends IsolatedRpcEndpoint with Logging {
```

Comment:
Actually, I'm wondering whether `BlockManagerMasterHeartbeatEndpoint ` should be an `IsolatedRpcEndpoint`. As `IsolatedRpcEndpoint` is mainly designed for those heavy&busy endpoints, e.g. `BlockManagerMasterEndpoint`,  `DriverEndpoint`, while `BlockManagerMasterHeartbeatEndpoint` doesn't seems to be high load. cc @vanzin @squito

Reply:
I just copy from `BlockManagerMasterEndpoint` :) 
`IsolatedRpcEndpoint` was merged recently. I miss the background about this.
Do you suggest to change back to `ThreadSafeRpcEndpoint`?

Reply:
Yes, I think so.

Reply:
Changed